packages/specs/json-mapper/src/utils/serialize.ts

```diff
@@ -65,7 +65,10 @@ export function classToPlainObject(obj: any, options: JsonSerializerOptions<any,
 
   const entity = JsonEntityStore.from(type || obj);
 
-  return getSchemaProperties(entity, obj).reduce((newObj, [key, propStore]) => {
+  const additionalProperties = !!entity.schema.get("additionalProperties");
+  const schemaProperties = getSchemaProperties(entity, obj);
+
```

Comment:
```typescript
const additionalProperties = !!entity.schema.get("additionalProperties");
const schemaProperties = getSchemaProperties(entity, obj);
const properties = new Set<string>()
const out: any = schemaProperties.reduce((newObj, [key, propStore]) => {
    properties.add(key);
    
    // ...
})

 if (additionalProperties) {
    objectKeys(obj).forEach((key: any) => {
       if (!properties.has(key)) {
          out[key] = obj[key];
       }
    });
  }
```

Reply:
This one gives me an error because the 'key' parameter in reduce is typed as 'string | number | symbol' :
<img width="713" alt="image" src="https://user-images.githubusercontent.com/58047787/153824210-b67a5006-645b-403f-86ba-369d452d8ec4.png">
I can solve this in one of 3 ways:
```suggestion
const properties = new Set<any>()
```
or
```suggestion
properties.add(key as string);
```
or
```suggestion
properties.add(key.toString());
```

Which do you prefer ?

Reply:
properties.add(key as string);

is fine for me :)